astropy/coordinates/geodetic.py

```diff
@@ -0,0 +1,136 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import erfa
+
+from astropy import units as u
+from astropy.utils.decorators import format_doc
+
+from .angles import Latitude, Longitude
+from .representation import (
+    BaseRepresentation,
+    CartesianRepresentation,
+)
+
+__all__ = [
+    "BaseGeodeticRepresentation",
+    "WGS84GeodeticRepresentation",
+    "WGS72GeodeticRepresentation",
+    "GRS80GeodeticRepresentation",
+]
+
+ELLIPSOIDS = {}
+"""Available ellipsoids (defined in erfam.h, with numbers exposed in erfa)."""
+# Note: they get filled by the creation of the geodetic classes.
+
+
+def _check_ellipsoid(ellipsoid=None, default="WGS84"):
+    if ellipsoid is None:
+        ellipsoid = default
+    if ellipsoid not in ELLIPSOIDS:
+        raise ValueError(f"Ellipsoid {ellipsoid} not among known ones ({ELLIPSOIDS})")
+    return ellipsoid
+
+
+geodetic_base_doc = """{__doc__}
+
+    Parameters
+    ----------
+    lon, lat : angle-like
+        The longitude and latitude of the point(s), in angular units. The
+        latitude should be between -90 and 90 degrees, and the longitude will
+        be wrapped to an angle between 0 and 360 degrees. These can also be
+        instances of `~astropy.coordinates.Angle` and either
+        `~astropy.coordinates.Longitude` not `~astropy.coordinates.Latitude`,
+        depending on the parameter.
+
+    height : `~astropy.units.Quantity` ['length']
+        The height to the point(s).
+
+    copy : bool, optional
+        If `True` (default), arrays will be copied. If `False`, arrays will
+        be references, though possibly broadcast to ensure matching shapes.
+"""
+
+
+@format_doc(geodetic_base_doc)
+class BaseGeodeticRepresentation(BaseRepresentation):
+    """
+    Base class for geodetic representations.
+    Subclasses need to set attributes ``_equatorial_radius`` and ``_flattening``
+    to quantities holding correct values (with units of length and dimensionless,
+    respectively), or alternatively an ``_ellipsoid`` attribute to the relevant ERFA
+    index (as passed in to erfa.eform).
+    """
+
+    attr_classes = {"lon": Longitude, "lat": Latitude, "height": u.Quantity}
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
```

Comment:
Put the `super` call at the end, so that it gets executed only when the part here is successful, otherwise you register representations even if they do not initialize properly.